src/drivers/serac.cpp

```diff
@@ -126,39 +125,36 @@ int main(int argc, char *argv[])
     return 2;
   }
 
-  mesh = new mfem::Mesh(imesh, 1, 1, true);
+  auto mesh = std::make_unique<mfem::Mesh>(imesh, 1, 1, true);
   imesh.close();
 
-  // declare pointer to parallel mesh object
-  mfem::ParMesh *pmesh = NULL;
-
   // mesh refinement if specified in input
   for (int lev = 0; lev < ser_ref_levels; lev++) {
     mesh->UniformRefinement();
   }
 
-  pmesh = new mfem::ParMesh(MPI_COMM_WORLD, *mesh);
+  // create the parallel mesh
+  auto pmesh = std::make_shared<mfem::ParMesh>(MPI_COMM_WORLD, *mesh);
   for (int lev = 0; lev < par_ref_levels; lev++) {
     pmesh->UniformRefinement();
   }
 
-  delete mesh;
-
   int dim = pmesh->Dimension();
 
   // Define the solid solver object
   NonlinearSolidSolver solid_solver(order, pmesh);
 
   // Project the initial and reference configuration functions onto the
   // appropriate grid functions
-  mfem::VectorCoefficient *defo_coef = new mfem::VectorFunctionCoefficient(dim, InitialDeformation);
+  auto defo_coef = std::make_shared<mfem::VectorFunctionCoefficient>(dim, InitialDeformation);
 
   mfem::Vector velo(dim);
   velo = 0.0;
 
-  mfem::VectorConstantCoefficient *velo_coef = new mfem::VectorConstantCoefficient(velo);
+  auto velo_coef = std::make_shared<mfem::VectorConstantCoefficient>(velo);
 
-  std::vector<mfem::VectorCoefficient *> coefs = {defo_coef, velo_coef};
+  auto coefs = {std::static_pointer_cast<mfem::VectorCoefficient>(defo_coef),
+                std::static_pointer_cast<mfem::VectorCoefficient>(velo_coef)};
```

Comment:
Hmm, this is a little confusing-- how is a user supposed to know that displacement goes first and velocity second? Maybe we make the ProjectState implementation specific to the solver, e.g. 
`NonlinearSolidSolver::ProjectDisplacement()` and `NonlinearSolidSolver::ProjectVelocity()`.

I'm also not entirely sure what "Project" means to MFEM. That word has multiple meanings depending on the context. If this is just setting the initial displacement/velocity fields, we might be better off calling them
`NonlinearSolidSolver::SetDisplacement()` and `NonlinearSolidSolver::SetVelocity()`.

Reply:
Good idea! I'll fix this.